File: CSharp part2/03. Methods/04.AppearanceCount/AppearanceCount.cs
// Write a method that counts how many times given number appears in given array. Write a test program to check if the method is workings correctly.

using System;

class AppearanceCount
{
    static string CountAppearance(int[]array, int element)
    {
        int counter = 0;
        string result = "";
        foreach (int number in array)
        {
            if (number == element)
            {
                counter++;
            }
        }
        if (counter == 0)
        {
            result = "There is no such element";
        }
        else 
        {
            result = "Element "+element+" has appearance count = "+counter.ToString();
        }
        return result;
    }
    static void Main()
    {
        Console.WriteLine("We are given array with elements");
        int[] array = { 5, 10, 15, 20, 5, 10, 15, 5, 30 };
        foreach (int number in array)
        {
            Console.Write(number + ", ");
        }
        Console.Write("\nEnter an element from the array above:");
        int element = int.Parse(Console.ReadLine());
        Console.WriteLine(CountAppearance(array,element));
    }
}

File: JavaScript-Fundamentals/Functions/05.AppearanceCount/AppearanceCount.js
// Write a function that counts how many times given number appears in given array.
// Write a test function to check if the function is working correctly.

var jsConsole;

var arr = [4,5,6,2,9,5,7,0,3,2,4,5,6,7];

function findCount (target, array) {
	var count = 0;
	for(var i = 0; i < array.length; i+=1) {
		if(array[i] == target){
			count+=1;
		}
	}
	if (count === 0){
		return 'number has no occurences';
	} else {
		return 'Number:'+target+' Ocurrences:'+count;
	}
}

function test () {
	var firstTest = findCount(4,arr);
	var secondTest = findCount(3,arr);
	var thirdTest = findCount(6,arr);
	return firstTest + '\n' + secondTest + '\n' + thirdTest;
}

jsConsole.writeLine(findCount(5,arr));
jsConsole.writeLine('Test function results:');
jsConsole.writeLine(test());
